v2/metadata/client.go

```diff
@@ -101,12 +106,19 @@ func NewClient(serverAddress, serverPort string) (*Client, error) {
 
 // ExecutionConfig represents the input parameters and artifacts to an Execution.
 type ExecutionConfig struct {
+	TaskName         string
+	Name             string // optional, MLMD execution name. When provided, this needs to be unique among all MLMD executions.
+	ExecutionType    ExecutionType
+	ParentID         int64 // parent execution ID, when not provided, it means root.
 	InputParameters  map[string]*structpb.Value
 	InputArtifactIDs map[string][]int64
-	TaskName, PodName, PodUID, Namespace,
-	Image, CachedMLMDExecutionID, ExecutionType, FingerPrint string
-	// a temporary flag to special case some logic for root DAG
-	IsRootDAG bool
+
+	// ContainerExecution custom properties
+	Image, CachedMLMDExecutionID, FingerPrint string
+	PodName, PodUID, Namespace                string
+
+	// DAGExecution custom properties
+	PipelineRoot string
```

Comment:
Note, I'm storing PipelineRoot in DAG execution's custom property.
I'm still considering whether this is necessary, we may also just store PipelineRoot in pipeline run context, because each pod should get the context anyway.

Reply:
Updated to only store pipeline root in run context, it's enough.